tmt/checks/coredump.py

```diff
@@ -0,0 +1,350 @@
+                    )
+                ).stdout
+
+            if not output:
+                return False
+
+            # Process each crash entry
+            has_crashes = False
+            for line in output.splitlines():
+                fields = line.split()
+
+                pid = fields[4]
+                sig = fields[7]
+                corefile = fields[8]
+                exe = fields[9].replace("/", "_")
+
+                # Get detailed info for this crash
+                crash_info = guest.execute(
+                    Command("coredumpctl", "info", "--no-pager", pid)
+                    ).stdout
+
+                if not crash_info:
+                    logger.debug(f"No crash info available for PID {pid}")
+                    continue
+
+                # Skip if this crash matches any ignore pattern
+                if self.ignore_patterns:
+                    ignored = False
+                    for pattern in self.ignore_patterns:
+                        if pattern.search(crash_info):
+                            logger.debug(
+                                f"Ignoring crash due to pattern '{pattern.pattern}': {crash_info}"
+                                )
+                            ignored = True
+                            break
+                    if ignored:
+                        continue
+
+                # Save the crash info
+                info_file = f"{check_files_path}/dump.{exe}_{sig}_{pid}.txt"
+                guest.execute(
+                    Command("sh", "-c", f"coredumpctl info --no-pager {pid} > {info_file}")
+                    )
+                logger.debug(f"Saved crash info to {info_file}")
+
+                # Try to save the coredump if available
+                if corefile not in ("none", "missing"):
+                    dump_file = f"{check_files_path}/dump.{exe}_{sig}_{pid}.core"
+                    try:
+                        guest.execute(
+                            Command("coredumpctl", "dump", "--no-pager", "-o", dump_file, pid)
+                            )
+                        logger.debug(f"Saved coredump to {dump_file}")
+                    except tmt.utils.RunError as exc:
+                        logger.debug(f"Failed to save coredump for PID {pid}: {exc}")
+                else:
+                    logger.debug(f"Skipping dump for PID {pid}, corefile status: {corefile}")
+
+                # This is a non-ignored crash
+                has_crashes = True
+
+            return has_crashes
+
+        except tmt.utils.RunError as exc:
+            logger.debug(f"Failed to check for crashes: {exc}")
+            return False
+
+    def _check_coredump(
+            self, invocation: "TestInvocation", event: CheckEvent, logger: tmt.log.Logger
+            ) -> ResultOutcome:
+        """Check coredump status and return appropriate result."""
+        # Only configure coredump in before_test
```

Comment:
Maybe the branch below should be in its own method, e.g. `create_coredump_timestamp()`, which would be then called only before the test. Written like this, the method depends on external state and decides on its own what to do, and if there's a condition like this the outside world knows, why even giving this method a chance to not obey? `before_test` will call it, `after_test` will not.

Reply:
Addressed in c0c05df176604cf7fa3ec96d92f952e03b886726, though at some point I became quite confused about what should be where, so reviews needed :)